File: code/54197.pas
program Bai197;
  var n, x: integer;
  function TONGUOC(a: integer): integer;
    var i, s: integer;
  begin
  s := 0;
  for i := 1 to a div 2 do 
     if a mod i = 0 then  s:= s +i;
 TONGUOC := s;
  end;
  
  begin
  readln(n);
  for x := 1 to n do if TONGUOC(x) = x then write(x, '  ');
end.

File: code/54197.cpp
#include <iostream>
using namespace std;

int tonguoc(int n){
    int s,x;
    s=0;
    for (x=1; x<=n/2; x++){
        if (n%x==0)
            s=s+x;
    }
    return s;
}

int main() 
{
    int n;
    cin>>n;
    for (int x=1; x<=n; x++)
        if (tonguoc(x)==x)
            cout<<x<<" ";
    return 0;
}